src/indexing/test/lfs.cpp

```diff
@@ -0,0 +1,376 @@
+
+  constexpr size_t buf_size = 1 << 10;
+  auto inbound_buffer = std::make_unique<ringbuffer::TestBuffer>(buf_size);
+  ringbuffer::Reader inbound_reader(inbound_buffer->bd);
+  auto outbound_buffer = std::make_unique<ringbuffer::TestBuffer>(buf_size);
+  ringbuffer::Reader outbound_reader(outbound_buffer->bd);
+
+  asynchost::LFSFileHandler host_files(
+    std::make_shared<ringbuffer::Writer>(inbound_reader));
+  host_files.register_message_handlers(host_bp.get_dispatcher());
+
+  ccf::indexing::EnclaveLFSAccess enclave_lfs(
+    std::make_shared<ringbuffer::Writer>(outbound_reader));
+  enclave_lfs.register_message_handlers(enclave_bp.get_dispatcher());
+
+  ccf::indexing::LFSKey key_a("Blob A");
+  ccf::indexing::LFSContents blob_a{0, 1, 2, 3, 4, 5, 6, 7};
+
+  ccf::indexing::LFSKey key_b("Blob B");
+  ccf::indexing::LFSContents blob_b{'a', 'b', 'c'};
+
+  enclave_lfs.store(key_a, ccf::indexing::LFSContents(blob_a));
+  enclave_lfs.store(key_b, ccf::indexing::LFSContents(blob_b));
+
+  REQUIRE(2 == host_bp.read_all(outbound_reader));
+
+  {
+    INFO("Load entries");
+
+    auto result_a = enclave_lfs.fetch(key_a);
+    REQUIRE(result_a->fetch_result == ccf::indexing::FetchResult::Fetching);
+
+    auto result_b = enclave_lfs.fetch(key_b);
+    REQUIRE(result_b->fetch_result == ccf::indexing::FetchResult::Fetching);
+
+    host_bp.read_all(outbound_reader);
+    enclave_bp.read_all(inbound_reader);
+
+    REQUIRE(result_a->fetch_result == ccf::indexing::FetchResult::Loaded);
+    REQUIRE(result_a->contents == blob_a);
+
+    REQUIRE(result_b->fetch_result == ccf::indexing::FetchResult::Loaded);
+    REQUIRE(result_b->contents == blob_b);
+  }
+
+  {
+    INFO("Host cache provides wrong file");
+    REQUIRE(std::filesystem::copy_file(
+      host_files.root_dir /
+        ccf::indexing::EnclaveLFSAccess::obfuscate_key(key_b),
+      host_files.root_dir /
+        ccf::indexing::EnclaveLFSAccess::obfuscate_key(key_a),
+      std::filesystem::copy_options::overwrite_existing));
+
+    auto result = enclave_lfs.fetch(key_a);
+
+    host_bp.read_all(outbound_reader);
+    enclave_bp.read_all(inbound_reader);
+
+    REQUIRE(result->fetch_result == ccf::indexing::FetchResult::Corrupt);
+    REQUIRE(result->contents != blob_a);
+  }
+
+#ifndef PLAINTEXT_CACHE
+  {
+    INFO("Host cache provides corrupt file");
+    const auto b_path = host_files.root_dir /
+      ccf::indexing::EnclaveLFSAccess::obfuscate_key(key_b);
+    const auto original_b_contents = read_file(b_path);
+
+    for (auto i = 0; i < original_b_contents.size(); ++i)
+    {
+      write_file_corrupted_at(b_path, i, original_b_contents);
+
+      auto result = enclave_lfs.fetch(key_b);
+
+      host_bp.read_all(outbound_reader);
+      enclave_bp.read_all(inbound_reader);
+
+      REQUIRE(result->fetch_result == ccf::indexing::FetchResult::Corrupt);
+      REQUIRE(result->contents != blob_b);
+    }
+  }
+#endif
+}
+
+TEST_CASE("Integrated cache" * doctest::test_suite("lfs"))
```

Comment:
Second unit test, using the new `Strategy` that uses the LFS to offload entries. Again a lot of boiler-plate, since this also has the boiler-plate of the indexing system.